### ISSUE TITLE
Excluded schema is still included in code generation of foreign keys in PostgreSQL

### ISSUE THREAD

**Issue by jleberre** [jOOQ/jOOQ#10305]
During code generation JOOQ seems to be mixing up foreign keys from different schemas if the tables have the same names.

This DDL...

```SQL
CREATE SCHEMA schema1;

CREATE TABLE schema1.foo (
    id INTEGER CONSTRAINT foo_pk PRIMARY KEY
);

CREATE TABLE schema1.bar (
	id INTEGER CONSTRAINT bar_pk PRIMARY KEY,
	ref INTEGER CONSTRAINT my_fk REFERENCES schema1.foo(id)
);

create schema schema2;

CREATE TABLE schema2.foo (
	id INTEGER CONSTRAINT foo_pk PRIMARY KEY
);

CREATE TABLE schema2.bar (
	id INTEGER CONSTRAINT bar_pk PRIMARY KEY,
	ref INTEGER CONSTRAINT my_other_fk REFERENCES schema2.foo(id)
);
```
...with the following code-gen configuration...

```XML
<execution>
  <id>jooq-test</id>
  <phase>generate-sources</phase>
  <goals>
    <goal>generate</goal>
  </goals>
  <configuration>
    <generator>
      <database>
        <name>org.jooq.meta.postgres.PostgresDatabase</name>
        <includes>.*</includes>
        <inputSchema>schema1</inputSchema>
      </database>
      <target>
        <packageName>jooq.test.schema1</packageName>
        <directory>${project.build.directory}/generated-sources/jooq/schema1</directory>
      </target>
    </generator>
  </configuration>
</execution>
```
...will produce the following `Keys` class 

```Java
public class Keys {
   ...
    private static class ForeignKeys0 {
        public static final ForeignKey<BarRecord, FooRecord> BAR__MY_FK = Internal.createForeignKey(Keys.FOO_PK, Bar.BAR, "my_fk", new TableField[] { Bar.BAR.REF }, true);
        public static final ForeignKey<BarRecord, FooRecord> BAR__MY_OTHER_FK = Internal.createForeignKey(Keys.FOO_PK, Bar.BAR, "my_other_fk", new TableField[] { Bar.BAR.REF }, true);
    }
}
```
### Expected behavior
There should be only one foreign key (`BAR__MY_FK`) in the generated `Keys` class

### Actual behavior
Both foreign keys, from `schema1` and `schema2` appear in the generated `Keys` class

Looking at the logs...
```
[DEBUG] Fetched result           : +-----------+-------------+------------+-------------+-----------+-------------+------------+-------------+-------+-----------+-----------+-----------+-------+-------------+
[DEBUG]                          : |pktable_cat|pktable_schem|pktable_name|pkcolumn_name|fktable_cat|fktable_schem|fktable_name|fkcolumn_name|key_seq|update_rule|delete_rule|fk_name    |pk_name|deferrability|
[DEBUG]                          : +-----------+-------------+------------+-------------+-----------+-------------+------------+-------------+-------+-----------+-----------+-----------+-------+-------------+
[DEBUG]                          : |{null}     |schema1      |foo         |id           |{null}     |schema1      |bar         |ref          |      1|          3|          3|my_fk      |foo_pk |            7|
[DEBUG]                          : |{null}     |schema2      |foo         |id           |{null}     |schema2      |bar         |ref          |      1|          3|          3|my_other_fk|foo_pk |            7|
[DEBUG]                          : +-----------+-------------+------------+-------------+-----------+-------------+------------+-------------+-------+-----------+-----------+-----------+-------+-------------+
[DEBUG] Fetched row(s)           : 2
[INFO] Adding foreign key       : my_fk (schema1.bar.ref) referencing foo_pk
[INFO] Adding foreign key       : my_other_fk (schema1.bar.ref) referencing foo_pk
```

...we can see that the foreign key `my_other_fk` from `schema2` has been interpreted as a foreign key from `schema1`

The problem is even worse when the foreign keys have the same name in both schemas since it will lead to only one foreign key declared in the generated `Keys` class BUT with the field referenced **twice** (see example below) causing `ArrayIndexOutOfBoundsException` when joining tables with `onKey()`.
```Java
    private static class ForeignKeys0 {
        public static final ForeignKey<BarRecord, FooRecord> BAR__MY_FK = Internal.createForeignKey(Keys.FOO_PK, Bar.BAR, "my_fk", new TableField[] { Bar.BAR.REF, Bar.BAR.REF }, true);
    }
```
This unexpected behaviour seems to occur since version 3.12.0

### Versions:

- jOOQ: 3.13.2
- Java: OpenJDK 11
- Database (include vendor): PostgreSQL
- OS: Windows 10
- JDBC Driver (include name if inofficial driver):

**Comment by jleberre** [comment]
@lukaseder tell me if you need more informations.

**Comment by lukaseder** [comment]
Thank you very much for your patience, @jleberre. I'll be in touch should more information be needed.

**Comment by lukaseder** [comment]
I can reproduce this. Will investigate tomorrow.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10352)
